rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,308 @@
+of displaying to the user, but the link between that and any actual property of
+the trusted objects is completely up to the publisher, and based on trusting the
+publisher alone.~~
+
+As an example, technically there is nothing stopping Google from (perhaps at
+some point in the future) repurposing the manifest identified by
+`https://google.com/oak/manifests/count_people_from_pixels.json` to publish
+hashes of nodes that count the number of cats instead of the number of people.
+The only understanding of what the list of hashes published in a manifest "mean"
+is outside of the manifest system itself. It may include formal or informal
+guarantees about functionality or policies that the objects are subject to, but
+those need to be published separately.
+
+For instance, Google may have a separate web page in which expectations of what
+is published in each manifest are detailed, with links to each manifest, and if
+relevant perhaps also technical instructions for third parties to independently
+verify that such properties are upheld (e.g. source code, training data, etc.).
+None of this affects the functioning of the delegation mechanism in Oak though.
+
+## Information Flow Control and Declassification Privilege
+
+Oak is extended with a new principal referring to manifests. When a piece of
+data is labelled with the URL of a manifest, Oak treats it as a regular IFC
+label.
+
+The Oak Application format is extended to explicitly list the URLs of any
+manifests that will be used throughout the execution of the application. The Oak
+Runtime fetches those manifests at startup, and recursively denormalizes each
+manifest into the corresponding list of Wasm hashes.
+
+When an Oak Application creates a new Wasm Node, the Oak Runtime checks the hash
+of the Wasm module to instantiate, and if it correspond to one or more of the
+manifests, then it grants the newly created node the corresponding privileges.
+
+For instance, let us assume the following manifests:
+
+- `https://google.com/oak/manifests/count_people_from_pixels.json`:
+
+  ```json
+  {
+    "trusted": [
+      {
+        "type": "wasm_module",
+        "value": "0000000000000000000000000000000000000000"
+      },
+      {
+        "type": "wasm_module",
+        "value": "1111111111111111111111111111111111111111"
+      }
+    ]
+  }
+  ```
+
+- `https://apple.com/oak/manifests/count_people_from_pixels.json`:
+
+  ```json
+  {
+    "trusted": [
+      {
+        "type": "wasm_module",
+        "value": "1111111111111111111111111111111111111111"
+      },
+      {
+        "type": "wasm_module",
+        "value": "2222222222222222222222222222222222222222"
+      }
+    ]
+  }
+  ```
+
+When an Oak Node with hash `1111111111111111111111111111111111111111` is
+instantiated, it receives the declassification privilege of
+"manifest:https://google.com/oak/manifests/count_people_from_pixels.json ∧
+manifest:https://apple.com/oak/manifests/count_people_from_pixels.json ∧
```

Comment:
Let's use a general url instead of apple's?

Reply:
Done.